src/bun.js/node/node_fs.zig

```diff
@@ -5036,15 +5036,17 @@ pub const NodeFS = struct {
     fn readdirWithEntries(
         args: Arguments.Readdir,
         fd: bun.FileDescriptor,
+        basename: [:0]const u8,
         comptime ExpectedType: type,
         entries: *std.ArrayList(ExpectedType),
     ) Maybe(void) {
         const dir = fd.asDir();
         const is_u16 = comptime Environment.isWindows and (ExpectedType == bun.String or ExpectedType == Dirent);
-        var iterator = DirIterator.iterate(
-            dir,
-            comptime if (is_u16) .u16 else .u8,
-        );
+
+        const dirent_path = bun.String.createUTF8(basename);
```

Comment:
Let's only create this String when `ExpectedType` == `Dirent`

Reply:
and we can do it lazily. if there are no entries, we should not create this String at all.